astroquery/heasarc/core.py

```diff
@@ -218,6 +218,8 @@ def _parse_result(self, response, verbose=False):
         elif "Software error:" in response.text:
             raise InvalidQueryError("Unspecified error from HEASARC database. "
                                     "\nCheck error message: \n{!s}".format(response.text))
+        elif "NO MATCHING ROWS" in response.text:
+            raise InvalidQueryError("No matching rows were found in the query.")
```

Comment:
isn't `NoResultsWarning` a more appropriate solution here? I mean the query can be valid when no match was found

Reply:
yes, I was lazy and didn't look at what we had in the warning list

Reply:
no worries, I also just vaguely remembered that we have something for these cases.